velox/type/Conversions.h

```diff
@@ -438,7 +438,14 @@ struct Converter<TypeKind::DATE, void, TRUNCATE> {
 
   static T cast(const Timestamp& t, bool& nullOutput) {
     static const int32_t kSecsPerDay{86'400};
-    return Date(t.getSeconds() / kSecsPerDay);
+    auto seconds = t.getSeconds();
+    if (seconds >= 0 || seconds % kSecsPerDay == 0) {
+      return Date(seconds / kSecsPerDay);
+    }
+    // For division with negatives, minus 1 to compensate the discarded
```

Comment:
Would you mention this fix in the PR description? Should we add some test cases for cast?

Reply:
PR description updated. I believe this PR is good to go.

These `cast` cases are well covered in the unit tests for `data()` function (as the underlying implementation is exactly the same). Also I haven't found any unit tests for `cast` itself so not sure if it's necessary.

If it is really needed I would suggest adding them in a separate PR since this one is for the `data()` function.